graphql-ballerina/Module.md

```diff
@@ -0,0 +1,685 @@
+
+When a service type is returned from a `graphql:Service`, the returning service type should also follow the rules of the `graphql:Service`, explained above.
+
+```ballerina
+import ballerina/graphql;
+
+service graphql:Service /graphql on new graphql:Listener(4000) {
+    resource function get profile() returns Person {
+        return new("Walter White", 51);
+    }
+}
+
+service class Person {
+    private string name;
+    private int age;
+    
+    public function init(string name, int age) {
+        self.name = name;
+        self.age = age;
+    }
+    
+    resource function get name() returns string {
+        return self.name;
+    }
+    
+    resource function get age() returns int {
+        return self.age;
+    }
+}
+```
+
+This will generate the following schema:
+
+```
+type Query {
+    profile: Person!
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+
+```graphql
+query getProfile {
+    profile {
+        name
+    }
+}
+```
+
+Which will result in the following JSON:
+
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White"
+        }
+    }
+}
+```
+
+#### Arrays
+A GraphQL resource can return an array of the types mentioned above.
+When a resource is returning an array, the result will be a JSON array.
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get people() returns Person[] {
+    Person p1 = { name: "Walter White", age: 51 };
+    Person p2 = { name: "James Moriarty", age: 45 };
+    Person p3 = { name: "Tom Marvolo Riddle", age: 71 };
+    return [p1, p2, p3];
+}
+```
+
+This will generate the following schema:
+
+```
+type Query {
+    profile: [Person!]!
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+```
+{
+    people {
+        name
+    }
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "people": [
+            {
+                "name": "Walter White"
+            },
+            {
+                "name": "James Moriarty"
+            },
+            {
+                "name": "Tom Marvolo Riddle"
+            }
+        ]
+    }
+}
+```
+
+Note that each element in the array consists only of the required `name` field.
+
+
+#### Optional Types
+A Ballerina GraphQL resource can return an optional type. When the return value is `()`, the resulting field in the JSON will be `null`.
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get profile(int id) returns Person? {
+    if (id == 1) {
+        return { name: "Walter White", age: 51 };
+    }
+}
+```
+
+This will generate the following schema:
+
+```
+type Query {
+    profile: Person
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
```

Comment:
```suggestion
This can be queried using the following document:
```
Is it correct to say "the following document"? Or, should it be "the following code"? Please check all occurrences.

Reply:
Same as above